File: web/src/model/victron_controller/dashboard/from_0_2_0_soc-chart.ts
// @ts-nocheck
import {SocChart as v0_2_0_SocChart} from './v0_2_0/SocChart'
import {SocChart as dashboard_SocChart} from './SocChart'

export function convert__soc_chart__from__0_2_0(from: v0_2_0_SocChart): dashboard_SocChart {
    return new dashboard_SocChart (
        JSON.parse(JSON.stringify(from.history)),
        JSON.parse(JSON.stringify(from.projection)),
        from.now_epoch_ms,
        from.now_soc_pct,
        from.discharge_target_pct,
        from.charge_target_pct
    )
}

File: crates/dashboard-model/src/victron_controller/dashboard/from_0_2_0_soc_chart.rs


pub fn convert__soc_chart__from__0_2_0(from: &crate::victron_controller::dashboard::v0_2_0::soc_chart::SocChart) -> crate::victron_controller::dashboard::soc_chart::SocChart {
    crate::victron_controller::dashboard::soc_chart::SocChart {
        history: serde_json::from_value(serde_json::to_value(&from.history).unwrap()).unwrap(),
        projection: serde_json::from_value(serde_json::to_value(&from.projection).unwrap()).unwrap(),
        now_epoch_ms: from.now_epoch_ms.clone(),
        now_soc_pct: from.now_soc_pct.clone(),
        discharge_target_pct: from.discharge_target_pct.clone(),
        charge_target_pct: from.charge_target_pct.clone(),
    }
}